crates/ide/src/rename.rs

```diff
@@ -131,6 +160,46 @@ pub(crate) fn will_rename_file(
     Some(change)
 }
 
+// FIXME: Allow usage with `extern crate`?
+fn alias_fallback(
+    sema: &Semantics<'_, RootDatabase>,
+    syntax: &SyntaxNode,
+    FilePosition { file_id, offset }: FilePosition,
+    new_name: &str,
+) -> RenameResult<Option<SourceChange>> {
+    match sema.find_node_at_offset_with_descend::<ast::UseTree>(syntax, offset) {
+        Some(use_tree) => {
+            if let Some(_rename) = use_tree.rename() {
+                return Ok(None); // FIXME: path already has an alias
```

Comment:
I'd say we should just rename the existing alias in that case. It is a bit weird, but more consistent that way imo

Reply:
Yeah, seemed weird, but I guess consistency is more important.